### PR TITLE
Fixes #3078 - Newlines in markdown content

### LINKED ISSUE #3078: Bug in Model (Definition) Description with newline characters
I have the following (example from my json definitions) - Please note very long description with newline characters:

```
        "Polygon": {
            "type": "object",
            "description": "GeoJSon Polygon geometry.\nExample:\n{\n    \"type\": \"Polygon\",\n    \"coordinates\": [\n        [\n            [\n                19.01064708,\n                -33.65407188\n            ],\n            [\n              19.01067012,\n              -33.65400312\n            ],\n            [\n              19.01083032,\n              -33.65394183\n            ],\n            [\n              19.01098944,\n              -33.65414082\n            ],\n            [\n              19.01106252,\n              -33.65423181\n            ],\n            [\n              19.01085048,\n              -33.65432082\n            ],\n            [\n              19.0107036,\n              -33.65414091\n            ],\n            [\n              19.01064708,\n              -33.65407188\n            ]\n        ]\n    ]\n}",
            "properties": {
                "type": {
                    "type": "string",
                    "enum": [
                        "Polygon"
                    ],
                    "description": "the geometry type - Polygon"
                },
                "coordinates": {
                    "type": "array",
                    "items": {
                        "type": "array",
                        "items": {
                            "$ref": "#/definitions/Point2D"
                        }
                    }
                }
            }
        },
```

Currently the newline character is ignored in the json description and renders as follow:

image

By giving the td (where the description lies) a style of: 
`white-space: pre;`

it renders correctly:

image

### PR DESCRIPTION
**Pull request by owenconti**
Fixes #3078 

Added `breaks: true` to Remarkable so newlines are rendered as line breaks. Remove `margin-top` from `<p>` tags within `.model` elements to fix the alignment in the model's description text.

Ping @webron @bodnia @shockey 

screen shot 2017-07-08 at 10 41 54 am

screen shot 2017-07-08 at 10 42 52 am

screen shot 2017-07-08 at 10 47 21 am

### CODE REVIEW

**Comment by webron** [comment]
@owenconti testing the example in #3078, the rendering is still off - the indentation is being lost.

It should look more like:
```
GeoJSon Polygon geometry.
Example:
{
    \"type\": \"Polygon\",
    \"coordinates\": [
        [
            [
                19.01064708,
                -33.65407188
            ],
            [
```

but in practice they all start at the same level.

**Comment by owenconti** [comment]
@webron updated. Implemented fix as per @1Map's suggestion

**Comment by owenconti** [comment]
Okay, I can't figure out how we can preserve whitespace. It seems the underlying Remarkable library we are using is stripping the whitespace: https://github.com/jonschlinkert/remarkable

My recommendation is: If you want code in your descriptions, wrap the code in markdown's '```'

**Comment by webron** [comment]
That actually might make sense. Markdown is a superset of HTML - meaning as long as there are no markdown or direct HTML tags, the text would be treated as plain text inside an HTML document, and as far as I recall, that means whitespaces get lost?

**Comment by shockey** [comment]
Yes, generally in HTML/CSS [sequences of whitespace are collapsed](https://developer.mozilla.org/en-US/docs/Web/CSS/white-space). 

I'm in favor of sticking with the underlying specifications here.

**Comment by webron** [comment]
If that's the case then there's nothing to add/change in this PR. Functionality-wise, it can be merged.

File: src/core/components/providers/markdown.jsx [hunk 1]
```diff
@@ -18,10 +18,12 @@ function Markdown({ source }) {
...
   }
 
-  return <Remarkable
-    options={{html: true, typographer: true, linkify: true, linkTarget: "_blank"}}
-    source={sanitized}
-    ></Remarkable>
+  return <div className="markdown">
+    <Remarkable
+      options={{html: true, typographer: true, breaks: true, linkify: true, linkTarget: "_blank"}}
+      source={source}
```
Comment (shockey) [inline src/core/components/providers/markdown.jsx]: @owenconti, please make sure to not remove the markdown sanitization 😄

**Pull request merged**